features/atcryptoauth/atca_app.cpp

```diff
@@ -0,0 +1,304 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ATCA.h"
+#include "samples.h"
+#include "mbedtls/pk_info.h"
+
+#if defined (MBED_CONF_ATCAECC_APP_ENABLE)
+
+/** Encode ECC Public key in ASN.1 format.
+ *
+ *  @param ecc_pk       ECC Public key input.
+ *  @param asn_out      ASN.1 out buffer.
+ *  @param asn_out_len  Out buffer length.
+ *  @return             0 on success, -1 on failure
+ */
+int ecc_key_to_asn1( uint8_t * ecc_pk, uint8_t * asn_out, size_t asn_len, size_t * asn_out_len )
+{
+    int ret = 0;
+    mbedtls_ecp_keypair ecp_key;
+    mbedtls_ecp_keypair_init(&ecp_key);
+
+    if (mbedtls_ecp_group_load(&ecp_key.grp, MBEDTLS_ECP_DP_SECP256R1) != 0)
+    {
+        goto cleanup;
+    }
+    MBEDTLS_MPI_CHK( mbedtls_mpi_read_binary( &ecp_key.Q.X, ecc_pk, 32 ) );
+    MBEDTLS_MPI_CHK( mbedtls_mpi_read_binary( &ecp_key.Q.Y, ecc_pk + 32, 32 ) );
+    MBEDTLS_MPI_CHK( mbedtls_mpi_lset( &ecp_key.Q.Z, 1 ) );
+    if( mbedtls_ecp_point_write_binary( &ecp_key.grp, &ecp_key.Q,
+                MBEDTLS_ECP_PF_UNCOMPRESSED, asn_out_len, asn_out, asn_len ) != 0 )
+    {
+        goto cleanup;
+    }
+    return( 0 );
+cleanup:
+    return( -1 );
+}
+
+
+/** Handle request over serial interface from host certificate creation app.
```

Comment:
Note: I presume this talks with https://github.com/ARMmbed/mbedtls/pull/1360/files#diff-17c563c558cf11e74eb29973c635e7bd